server/src/main/java/org/elasticsearch/common/xcontent/XContentFieldFilter.java

```diff
@@ -73,8 +73,8 @@ static XContentFieldFilter newFieldFilter(String[] includes, String[] excludes)
             };
         } else {
             final XContentParserConfiguration parserConfig = XContentParserConfiguration.EMPTY.withFiltering(
-                Set.of(includes),
-                Set.of(excludes),
+                Set.copyOf(Arrays.asList(includes)),
+                Set.copyOf(Arrays.asList(excludes)),
```

Comment:
I had a chance to review it only after the pr was merged, may be we could consider this for a followup or a future change:

May be we should have a dedicated method for this in CollectionUtils so that intention is clear and nobody revert/simplify code to the shorter `Set.of` version. May be something like `CollectionUtils.setOfCollectionWithPossibleDuplicates(Collection<T> c)`

Reply:
not advocating against your suggestion, but we do have tests for this now so reverting the change would cause test failures

Reply:
This PR introduced a test for duplicate filters. I think that's what you want, right?

Reply:
Correct, the test would catch the revert. I was suggesting to have a dedicated set factory that would display the intention (create set from collection that may contain duplicates)